core/src/main/java/org/axonframework/eventhandling/saga/metamodel/AnnotationSagaMetaModelFactory.java

```diff
@@ -87,12 +90,9 @@ public InspectedSagaModel(List<MessageHandlingMember<? super T>> handlers) {
         }
 
         @Override
-        @SuppressWarnings("unchecked")
-        public List<SagaMethodMessageHandlingMember<T>> findHandlerMethods(EventMessage<?> eventMessage) {
+        public List<MessageHandlingMember<? super T>> findHandlerMethods(EventMessage<?> eventMessage) {
             return handlers.stream().filter(h -> h.canHandle(eventMessage))
-                           .map(h -> (SagaMethodMessageHandlingMember<T>) h.unwrap(SagaMethodMessageHandlingMember.class)
-                                                                           .orElse(null))
-                           .filter(Objects::nonNull).collect(Collectors.toCollection(ArrayList::new));
+                           .collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Isn't here simple `toList()` sufficient? This looks unnecessary verbose.

Reply:
Ah yeah good one. I just removed some function here, without adjusting the actual call.